cpp/src/lance/format/schema.cc

```diff
@@ -516,6 +558,25 @@ ::arrow::Result<std::shared_ptr<Schema>> Schema::Project(
   return Project(columns);
 }
 
+::arrow::Result<std::shared_ptr<Schema>> Schema::Merge(const Schema& other) const {
+  auto schema = std::make_shared<Schema>();
+  for (auto& field : fields()) {
+    auto existing_field = other.GetField(field->name_);
+    if (existing_field) {
+      ARROW_ASSIGN_OR_RAISE(auto merged, field->Merge(*existing_field));
+      schema->AddField(merged);
+    } else {
+      schema->AddField(field->Copy(true));
+    }
+  }
+  for (auto& other_field : other.fields()) {
+    if (schema->GetField(other_field->name_) == nullptr) {
+      schema->AddField(other_field->Copy(true));
+    }
+  }
+  return schema;
```

Comment:
just confirming: ExtensionType's would be considered non-nested right?

Reply:
Yes, it treats ext type as one single unit.